### PR TITLE
[Merged by Bors] - feat: enable fuel

### PR DESCRIPTION
**Pull request by davidbeesley**

### CODE REVIEW

**Review by morenol — APPROVED** [review]

File: crates/fluvio-smartengine/src/engine.rs [hunk 1]
```diff
@@ -24,7 +24,9 @@ pub struct SmartEngine(Engine);
 #[allow(clippy::new_without_default)]
 impl SmartEngine {
     pub fn new() -> Self {
-        Self(Engine::default())
+        let mut config = wasmtime::Config::default();
+        config.consume_fuel(true);
+        Self(Engine::new(&config).expect("Config is static"))
```
Comment (sehz) [inline crates/fluvio-smartengine/src/engine.rs:29]: should remove expect

**Review by sehz — CHANGES_REQUESTED** [review]

File: crates/fluvio-smartengine/src/engine.rs (hunk 1 above)
Comment (davidbeesley) [reply crates/fluvio-smartengine/src/engine.rs:29]: This expect can be tested before deploy.

Comment (davidbeesley) [reply crates/fluvio-smartengine/src/engine.rs:29]: As the config is static.

Comment (davidbeesley) [reply crates/fluvio-smartengine/src/engine.rs:29]: So success of Engine::new().expect() is unit tested.

Comment (davidbeesley) [reply crates/fluvio-smartengine/src/engine.rs:29]: the default() that was being called previously had an unwrap() inside anyway
```
impl Default for Engine {
    fn default() -> Engine {
        Engine::new(&Config::default()).unwrap()
    }
}
```

Comment (davidbeesley) [reply crates/fluvio-smartengine/src/engine.rs:29]: Yes but it can't fail.. it's a static configuration

Comment (davidbeesley) [reply crates/fluvio-smartengine/src/engine.rs:29]: Which is unit tested

Comment (sehz) [reply crates/fluvio-smartengine/src/engine.rs:29]: Fair enough

**Review by sehz — APPROVED** [review]

**Pull request closed without merging**